src/error.ts

```diff
@@ -121,6 +121,13 @@ export function createError<DataT = unknown>(
     err.statusCode = sanitizeStatusCode(input.statusCode, err.statusCode);
   } else if (input.status) {
     err.statusCode = sanitizeStatusCode(input.status, err.statusCode);
+  } else if (
+    typeof input.cause === "object" &&
```

Comment:
Since `cause` can have all sorts of shapes, I find it best to do some type checking here before we try and parse from it. If you prefer a different style on this kind of check (i.e. an external helper function), let me know and I'd be happy to refactor.

Reply:
Inline check is fine only `cause` can be `null` too (also saw it in https://github.com/apollographql/apollo-client/pull/11902). `typeof null === "object"` and well next line fails so we can simplify it to truehy checking like `input.cause?.statusCode`.